creator/ingest_runs/common/model.py

```diff
@@ -0,0 +1,182 @@
+import hashlib
+import uuid
+import re
+
+import django_rq
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+from django_fsm import FSMField, transition
+
+from creator.jobs.models import JobLog
+from creator.utils import stop_job
+
+User = get_user_model()
+
+
+def camel_to_snake(camel_str):
+    """
+    Convert CamelCase to snake_case
+    """
+    return re.sub(r'(?<!^)(?=[A-Z])', '_', camel_str).lower()
+
+
+def hash_versions(versions):
+    """
+    Uniquely identify a set of file Versions by computing an MD5 hash digest
+    using the primary keys of the Versions
+    """
+    version_id_str = "".join(sorted(v.kf_id for v in versions))
+    return hashlib.md5(version_id_str.encode("utf-8")).hexdigest()
+
+
+class State(object):
+    NOT_STARTED = "not_started"
+    RUNNING = "running"
+    FAILED = "failed"
+    CANCELED = "canceled"
+    COMPLETED = "completed"
+
+
+class IngestProcess(models.Model):
+    """
+    Common model functionality for ingest processes
+    (e.g. ingest run, validation run)
+    """
+    __queue__ = "ingest"
```

Comment:
Would this be better as an app setting? I think you also want to add the queue to `RQ_QUEUES`